decode/data.ts

```diff
@@ -1,23 +1,76 @@
-import { Event, IDecodedEvent, IPageData, IPingData, ITagData, Token } from "../types/data";
+import { Envelope, Event, MetricData, PageData, PingData, State, SummaryData, TagData, Token, Upload } from "../types/data";
+import { DataEvent, SummaryEvent } from "../types/decode";
 
-export default function(tokens: Token[]): IDecodedEvent {
+let summaries: { [key: number]: SummaryData[] } = null;
+const SUMMARY_THRESHOLD = 250;
+
+export function reset(): void {
+    summaries = {};
+}
+
+export function decode(tokens: Token[]): DataEvent {
     let time = tokens[0] as number;
     let event = tokens[1] as Event;
     switch (event) {
         case Event.Page:
-            let page: IPageData = {
+            let page: PageData = {
                 timestamp: tokens[2] as number,
```

Comment:
>timestamp [](start = 16, length = 9)

What are these fields (timestamp and elapsed)?

Reply:
timestamp is absolute unix time. The idea is that if we needed to, we could turn relative "time" values in payload to absolute ones. I've removed elapsed since it was adding confusion and I already have "time" in the event wrapper.